### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by stdavis**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+indent_size = 4
+
 [*]
 indent_style = space
-indent_size = 4
 charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
-[*.yml]
-indent_size = 2
```
Comment (steveoh) [inline .editorconfig:10]: **question**: isn't this useful for the ci parts of this project?

**Review by steveoh — COMMENTED** [review]
🎨

File: setup.py [hunk 2]
```diff
@@ -41,22 +44,28 @@ def read(*names, **kwargs):
...
-        "develop": [
-            "yapf",
-            "pylint",
+            "pytest-cov==5.*",
+            "pytest-instafail==0.5.*",
+            "pytest-mock==3.*",
+            "pytest-ruff==0.*",
+            "pytest-watch==4.*",
+            "pytest==8.*",
+            "black==24.*",
```
Comment (steveoh) [inline setup.py:63]:
```suggestion
```

**Comment by stdavis** [comment]
> .python-version can probably go to 11 right?

I believe that Pro is still at 3.9 and this project uses arcpy.

File: .editorconfig (hunk 1 above)
Comment (stdavis) [reply .editorconfig:10]: I don't think that it's needed. VSCode automatically detects that they are two spaces.

File: readme.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ A normalized string representing the entire address that was passed into the con
 <!-- Current conda install arcpy -c esri seems to be wonky; just clone to be safe -->
```
[the remark this answers is not in the document]
Comment (stdavis) [reply readme.md:102]: Yes. I tried and it failed to install.

File: .vscode/extensions.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+    // See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
+    // Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
+    // List of extensions which should be recommended for users of this workspace.
+    "recommendations": [
+        "editorconfig.editorconfig",
+        "njpwerner.autodocstring",
+        "ms-python.black-formatter",
```
[the remark this answers is not in the document]
Comment (stdavis) [reply .vscode/extensions.json:8]: We may want to make these changes in agrc/python. I was just copying from that project. And by "we", I mean you. ;)

**Comment by steveoh** [comment]
> I believe that Pro is still at 3.9 and this project uses arcpy.

oh right, i was thinking with 3.3 around the corner

File: .vscode/extensions.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (steveoh) [reply .vscode/extensions.json:8]: I'll make them to that project if you make them to this one :)

[the remark this answers is not in the document]
Comment (stdavis) [reply .vscode/extensions.json:8]: I'll take care of agrc/python. I'll include this with a few other changes.

**Review by steveoh — APPROVED** [review]

**Pull request merged**